### PR TITLE
feat: add productpage

### PR DESCRIPTION
**Pull request by sagargse**
#### Added Features Like :
* Sort By Price
* Include Out of Stock 
* Product Category

image

Preview Link : https://deploy-preview-3--ecommerce-kidszilla.netlify.app/

### CODE REVIEW

File: src/components/filter/Filter.jsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                            id="Flexible toys"
+                            checked={state.flexibleToys}
+                            onChange={() => dispatch({ type: "FLEXIBLE_TOYS" })}
+                        />
+                        <label for="toys">Flexible Toys</label>
+                    </div>
+                </div>
+                <hr />
+                <div className="filtertype-body-container">
+                    <h4>FILTER TYPE</h4>
```
Comment (khushijohri2001) [inline src/components/filter/Filter.jsx:80]: You can also give a filter according to rating. You have given ratings on the products. I would suggest you to add rating filter too.

File: src/components/filter/filter.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+.sort-by-category{
+    display: flex;
+    gap:2rem;
+}
+.filter-body-container .sort-by-price {
+    display: flex;
+    gap:2rem;
+}
+.sort-body-container,.category-body-container,.filtertype-body-container{
```
Comment (khushijohri2001) [inline src/components/filter/filter.css:24]: Nice that you have used the basics of CSS and not repeated the code for each class.

**Review by khushijohri2001 — APPROVED** [review]
All-in-all the code is clean!!
Liked it👍

File: src/components/filter/Filter.jsx (hunk 1 above)
Comment (sagargse) [reply src/components/filter/Filter.jsx:80]: okay will add that

File: src/components/filter/filter.css (hunk 2 above)
Comment (sagargse) [reply src/components/filter/filter.css:24]: > All-in-all the code is clean!! Liked it👍

Thanks

**Review by Rockyraven — APPROVED** [review]
your code is clean and easy to read 
you can go ahead

**Comment by sagargse** [comment]
> your code is clean and easy to read you can go ahead

Thanks!

**Pull request merged**